src/main/java/org/cirdles/topsoil/chart/JavaScriptChart.java

```diff
@@ -307,17 +306,19 @@ public void setData(VariableContext variableContext) {
         // of undefined objects on the other side of things
         afterInitialization(() -> {
             topsoil.call("clearData"); // old data must be cleared away
-
+            
             variableContext.getDataset().getEntries().forEach(entry -> {
-                JSObject row = (JSObject) webEngine.executeScript("new Object()");
+                if(entry.getIsSelected()==true) {
```

Comment:
Note that `if (entry.getIsSelected()) {` will always be better than this. In this case, however, use `Stream#filter` instead. Example:

``` java
variableContext.getDataset().getEntries().stream()
        .filter(Entry::getIsSelected) // although I don't like that this method exists
        .forEach(entry -> {
            // ...
        });
```